src/hostapi/jack/pa_jack.c

```diff
@@ -584,11 +636,13 @@ static PaError BuildDeviceList( PaJackHostApiRepresentation *jackApi )
 
         /* To determine how many input and output channels are available,
          * we re-query jackd with more specific parameters. */
-
-        sprintf( regex_pattern, "%s:.*", client_names[client_index] );
+        strcpy( device_name_regex_escaped, client_names[client_index] );
```

Comment:
strcpy is not save, because it does no length check. A one stop shopping function from the standard library is snprintf() with "%s".